### PR TITLE
Add tinkerbell 1.24 and 1.23 tests

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*
#3491 

*Description of changes:*
Add tinkerbell 1.24 and 1.23 tests

*Testing (if applicable):*
- make e2e

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/tinkerbell_test.go [hunk 1]
```diff
@@ -19,13 +20,14 @@ const (
 	nodeGroupLabel2 = "md-1"
 )
 
-func TestTinkerbellKubernetes122WithNodesPoweredOn(t *testing.T) {
+func TestTinkerbellKubernetes124WithNodesPoweredOn(t *testing.T) {
```
Comment (pokearu) [inline test/e2e/tinkerbell_test.go:23]: Why are we replacing the 1.22 tests?

**Review by pokearu — COMMENTED** [review]
Looks good 👍  Just a few questions.
1. Do we want to add BottleRocket tests for 1.24 as well?
2. Did we update our CI for `tinkerbell_ci:image_ubuntu_1_24` and `tinkerbell_ci:image_redhat_1_24` ?

**Comment by mitalipaygude** [comment]
> Looks good 👍 Just a few questions.
> 
> 1. Do we want to add BottleRocket tests for 1.24 as well?
- yes we do, I am still working on those too. will be a different pr though.

> 2. Did we update our CI for `tinkerbell_ci:image_ubuntu_1_24` and `tinkerbell_ci:image_redhat_1_24` ?
- yes, the secrets have been created in the secrets manager for tinkerbell.

**Pull request merged**